packages/front-end/components/Experiment/ImportExperimentList.tsx

```diff
@@ -213,7 +213,22 @@ const ImportExperimentList: FC<{
                 </Link>
               </p>
             )}
-            <span>View Queries (below) for more information.</span>
+            <span>
+              <ViewAsyncQueriesButton
+                queries={
+                  data.experiments.queries?.length > 0
+                    ? data.experiments.queries.map((q) => q.query)
+                    : []
+                }
```

Comment:
hello :wave: 

maybe we can improve this algotithm

the map always return a empty array if it has no length.

example 1:

```js
[].map(q => q.query)
// will return []
```

example 2:

```js
const data = {
    experiments: {
        queries: []
    }
}

data.experiments.queries.map(q => q.query)
// will return []
```

---

it means that you don't need do the ternary operator. 

```js
queries={data.experiments.queries.map((q) => q.query)}
```

probably will work fine. please, test if works as expected before update you code.

---

_ps: also you can create a memoization with [useMemo](https://beta.reactjs.org/reference/react/useMemo) to improve the performance. *ignore* if you arent familiar with react hooks_



thanks you :)

Reply:
Sure, addressed